addons/sale/models/sale_order.py

```diff
@@ -849,31 +849,40 @@ def _show_cancel_wizard(self):
         return False
 
     def _find_mail_template(self, force_confirmation_template=False):
-        template_id = False
+        """ Get the appropriate mail template for the current sale's order(s) based on their state.
 
-        if force_confirmation_template or (self.state == 'sale' and not self.env.context.get('proforma', False)):
-            template_id = int(self.env['ir.config_parameter'].sudo().get_param('sale.default_confirmation_template'))
-            template_id = self.env['mail.template'].search([('id', '=', template_id)]).id
-            if not template_id:
-                template_id = self.env['ir.model.data'].xmlid_to_res_id('sale.mail_template_sale_confirmation', raise_if_not_found=False)
-        if not template_id:
-            template_id = self.env['ir.model.data'].xmlid_to_res_id('sale.email_template_edi_sale', raise_if_not_found=False)
+        If all SOs are 'sale' or there is the ``force_confirmation_template``,
+        the mail template is that of the sale confirmation.
+        Else, it returns the quotation's email template.
 
-        return template_id
+        :param bool force_confirmation_template: Allows to force the confirmation template to be
+        used in any case
+        :return: The correct mail template based on the current status
+        :rtype: record of `mail.template`
+        """
+        if force_confirmation_template or (
+            all(order.state == 'sale' for order in self) and not self.env.context.get('proforma')
+        ):
+            default_template_id = int(
+                self.env['ir.config_parameter'].sudo()
+                                               .get_param('sale.default_confirmation_template')
+            )
+            default_template = self.env['mail.template'].browse(default_template_id)
+            return default_template or self.env.ref('sale.mail_template_sale_confirmation')
+        return self.env.ref('sale.email_template_edi_sale')
```

Comment:
@Demesmaeker Can you review this change? I added back an improvement that we've made in our first task #68965.

Reply:
Ok for me, maybe add _sudo in the variable name, as the fact that it's sudo will stay (it's a tip Antoine gave me some days ago). It's better to put it in the name, so we don't forget about it.